src/connection_pool.rs

```diff
@@ -42,6 +42,35 @@ impl ConnectionPool {
         }
     }
 
+    pub fn has(&self, addr: &SocketAddr) -> bool {
+        let mut store = self.store.lock().unwrap_or_else(PoisonError::into_inner);
+
+        // Efficiently fetch the first entry whose key is equal to `key` and check if it exists
+        store
+            .map
+            .range_mut(Key::min(*addr)..=Key::max(*addr))
+            .next()
+            .is_some()
+    }
+
+    pub fn remove(&mut self, addr: &SocketAddr) -> Option<(quinn::Connection, ConnectionRemover)> {
```

Comment:
There is no point in returning the `Remover` here. The remover is a handle to remove the connection from the pool. But you are doing that manually in this function, so the remover has no purpose (calling `remove` on it would have no effect).

Reply:
Also, this removes only one connection with that address, but in some cases (*) there can be more than one. Wonder if it wouldn't make more sense to remove all of them instead?

*) if two peers open a connection to each other around the same time, they can end up having two connections to the same peer in the pool (one incoming and one outgoing).